app/public/yaml.js

```diff
@@ -7,38 +7,81 @@ window.addEventListener("load", function() {
 		return Array.prototype.find.call(collection, condition);
 	}
 
-	function buildName(string, ol, nameRegexp) {
-		return string.replace(nameRegexp, ol.dataset.key + "[" + ol.children.length + "]");
+	function nameRegexp(ol) {
+		return new RegExp(ol.dataset.key.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&") + "\\[\\d+\\]");
 	}
 
-	function hydrateLi(nameRegexp, li, i) {
-		var ol = li.parentElement;
-		var button = find(li.children, function(el) { return el.className === "clone"; });
+	function buildName(string, ol, number) {
+		return string.replace(nameRegexp(ol), ol.dataset.key + "[" + number + "]");
+	}
+
+	function enableDisableMoveButtons(ol, li, number) {
+		find(li.children, function(el) { return el.className === "move-up"; }).disabled = number < 1;
+		find(li.children, function(el) { return el.className === "move-down"; }).disabled = number >= ol.children.length - 1;
+	}
+
+	function hydrateLi(li) {
+		function findOrCreateButton(className, label) {
+			var button = find(li.children, function(el) { return el.className === className; });
 
-		if(!button) {
-			button = document.createElement("button");
-			li.appendChild(button);
+			if(!button) {
+				button = document.createElement("button");
+				li.appendChild(button);
 
-			button.type = "button";
-			button.className = "clone";
-			button.textContent = "Clone";
+				button.type = "button";
+				button.className = className;
+				button.textContent = label;
+			}
+
+			return button;
 		}
 
-		button.addEventListener("click", function() {
+		findOrCreateButton("clone", "Clone").addEventListener("click", function() {
 			var item = li.cloneNode(true);
-			renameAppendHydrate(ol, item, nameRegexp);
+			renameAppendHydrate(li.parentElement, item, nameRegexp(li.parentElement));
 		});
+
+		findOrCreateButton("move-up", "^").addEventListener("click", function() {
```

Comment:
Can we replace this text with either an SVG or more consistent characters? The sizing is pretty different between them now since one is ^ and one is V.